src/main/java/com/artipie/RepoConfig.java

```diff
@@ -98,6 +98,21 @@ public String type() {
         return this.string("type");
     }
 
+    /**
+     * Repo port.
+     * @return Repo port
+     */
+    public Optional<Integer> port() {
+        final int port = this.repoConfig().integer("port");
+        final Optional<Integer> empty;
+        if (port == -1) {
```

Comment:
@Sammers21 I'd suggest to read port as string and parse it in this method. Otherwise we pollute our code with dependency on library flaws (such as representing absent value as `-1`)